enclave-runtime/src/tls_ra/mocks.rs

```diff
@@ -0,0 +1,72 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use super::seal_handler::{SealStateAndKeys, UnsealStateAndKeys};
+use crate::error::Result as EnclaveResult;
+use itp_types::ShardIdentifier;
+use std::{
+	sync::{Arc, SgxRwLock as RwLock},
+	vec::Vec,
+};
+
+#[derive(Clone)]
+pub struct SealHandlerMock {
+	pub shielding_key: Arc<RwLock<Vec<u8>>>,
+	pub signing_key: Arc<RwLock<Vec<u8>>>,
+	pub state: Arc<RwLock<Vec<u8>>>,
```

Comment:
I'm guessing you're using `Arc`s here for the `#[derive(Clone)]`, because `RwLock` is not clonable? For the mock it's probably okay, but in a real implementation, this would most likely not be what we want. We'd have to implement clone ourselves and clone the inner value of `RwLock`.

Reply:
Agreed, but for the mock that was the easiest way to implement a test. At least for what I could see.